pkg/state/state.go

```diff
@@ -2625,13 +2642,6 @@ func (st *HelmState) timeoutFlags(release *ReleaseSpec) []string {
 
 func (st *HelmState) flagsForUpgrade(helm helmexec.Interface, release *ReleaseSpec, workerIndex int, opt *SyncOpts) ([]string, []string, error) {
 	flags := st.chartVersionFlags(release)
-
-	if release.Verify != nil && *release.Verify || release.Verify == nil && st.HelmDefaults.Verify {
```

Comment:
I guess we still need to pass the verify and keyring related flags to helm-upgrade, in case it isn't an OCI chart?

Reply:
@mumoshu good point. I will try to see the details.

Reply:
@yxxhero it will duplicate check for non-oci chart. is it ok?

Reply:
@yxxhero Thanks for checking it!

I think that's not OK, because I remember that the original issue was about `--verify` on local chart pre-pulled via`getOCIChart` fails (right?)

Maybe we need to check if it's an OCI chart or not before calling `flagsForUpgrade`. In `flagsForUpgrade`, we might need to add verify-and-keyring-related flags only if it's non-OCI.

WDYT?